### DISCUSSION TITLE
ERROR : insert into 'userGroups'('groupId','userId')

### DISCUSSION THREAD

**Discussion by lealzora** [requarks/wiki#5365] [Error / Bug Report]
The error occurs when using the "self-registration of user" functionality.

Actions done in settings:

1 - Access the Administration environment
2 - Access the "Authentication" module
3 - In the Active Strategies options select "LDAP/Active Directory"
4 - Apply strategy settings
5 - In the Registration options:
5.1 - Enable the "self-registration" option
5.2 - In the option "Assign to group", inform the group to which the user who registers himself wants to be linked

What is happening:

The user who self-registers, after informing the login data, an error message is displayed.
"**insert into 'userGroups'('groupId','userId') values ​​($1,$2),($3,$4),($5,$6) returning 'groupId' - insert or update on table 'userGroups' violates foreign key constraint 'usergroups_groupid_foreign'** "

Questions:
1 - Is there any solution to be delivered for future versions?
2 - Any contour suggestions?

**Comment by yysf-brait** [comment]
To provide some additional information: I did not select "LDAP/Active Directory", the same error still occured.
Unfortunately, I haven't found a solution yet
NEED HELP!  :(

**Comment by rafmcpaf** [comment]
Found the issue (I guess...) The Front-end selects by default a group with id = 0 (which does not exist). 
So when linking the new user (self-registration) to the groups, a group with id = 0 is linked as well. 
As this group does no exist, a foreign key exception is thrown. 

As a workaround, I manually ran the graphql-call to update the authentication strategies (f12), but removed the not existing group-id (prop autoEnrollGroups). After this call, all seems to be going fine!

**Comment by FilipeNas** [comment]
My workaround was to delete an entry on the **authentication table**, that reference a group that doesn't exists anymore.
I used a GUI tool to access Postgres, so i dont know the query to do that via command line.

**Comment by 1liminal1** [comment]
insert into "userGroups" ("groupId", "userId") values ($1, $2), ($3, $4) returning "userId" - insert or update on table "userGroups" violates foreign key constraint "usergroups_groupid_foreign"

Is there a way to get this working?

**Comment by Puuuudele** [comment]
we solved the problem as follows: 

We exposed the database port externally (in Docker Compose) with:

```
ports:
  - "5432:5432"
```

After that, we connected to the PostgreSQL database using pgAdmin (software available online). Under Schemas -> Tables -> authentication, we right-clicked on the authentication table and ran a query through Scripts -> Select. In the column "autoEnrollGroups", we set the value to `{"v":[21]}`. The number 21 in our case is the group ID of the default group. You can easily change the value in the software and save it to the database.

Afterward, I restarted the containers with `docker compose stop` and `docker compose up -d`, and the issue was resolved.
